### PR TITLE
Improve home page

### PR DESCRIPTION
**Pull request by MarcSkovMadsen**
New version of https://github.com/holoviz/panel/pull/4654 which after a rebase looked broken.

The improved home page will be deployed once in a while to https://pyviz-dev.github.io/panel/.

## Todo

- [x] Simplify Demo App by 1) Moving text to Description Tab 2) Renaming "Output" tab to "App" tab 3) Improve design.
- [] Get acceptance from tweeters to use their tweets (I have asked them).
- [] Get reviewed and finalized
- [] After finalization Philipp should upload thumbnails to holoviz cdn and update links.

## Nice to have

- *Featured Testimonials*: If we could get 4 testimonials from different types of trustworthy organisations or individuals it would be quite powerful @jbednar and others. If its not the sponsors or someone already visible on the home page it would help diversify our communication.
- *Featured Use Cases*: If we could find 4 great and public use cases that would also be very powerful @jbednar and others. If its not the sponsors or someone already visible on the home page it would help diversify our communication. Rapids CuxFilter could be one.

By *Featured* I mean adding them as panel similar to below

image

## Dependencies (Separate PRs).

- [] Philipp?: Get styling in order for for the "More", "Search" and "Theme" buttons in the header. See https://github.com/holoviz/panel/issues/4647
- [] Philipp?: Get Palmer Penguin app fixed. See https://github.com/holoviz/panel/issues/4670
- [] Philipp?: Fix FastTemplate css issues. See https://github.com/holoviz/panel/issues/4668
- [] Philipp?: Fix refresh issue. See https://github.com/holoviz/panel/issues/4682
- [] Sophia?: Contribute a 5 min *introduction to Panel* video to replace the current 30 mins from Marc from PyData CpH.
- [] Marc?: Contribute Portfolio Analyzer App to Featured apps. See #4683

The below would also be *nice to have*.

- [] ?: Reorder and Rename top entries in toc tree. See https://github.com/holoviz/panel/issues/4681
- [] Philipp? Add support for tooltips to nbsite. It would make a lot of sense to have tooltips on the thumbnails, so that users can be told what they lead to before clicking.
- [] Philipp? change link to Featured VideoStream app to a Pyoide version. This app works great in Pyodide and will give a great user experience there because it can update the videostream at 20 fps or more.
- [] Philipp? Make demo app width responsive. I could not do that.
- [] Philipp? : Make demo app work on mobile. Should probably use flexgrid or something to be able to show plot.

### CODE REVIEW

**Comment by MarcSkovMadsen** [comment]
I've added some featured videos. If you want to keep that section and content @philippjfr , you should upload the thumbnails to the CDN.

sophia_yang_chatbot

thu_vu

marc_skov_madsen_pydata_copenhagen_2022

sophia_yang_holoviz

I think we could benefit from having

- a very sharp 5 minute overview of Panel (and HoloViz) - Sophias HoloViz video plays that role but it could be updated: 
- a 10 minute walk though of build a simple Panel data app. My video and Thu Vus video plays that role.

Sophias video about chat bot communicates that you can develop state of the Art AI apps with Panel.

**Comment by MarcSkovMadsen** [comment]
I think the demo app could benefit from

- being width responsive?
- having removed most of the text as its now written in the docs it self.

image

**Comment by MarcSkovMadsen** [comment]
I think the Crossfiltering and OGGM Glaciers apps are very similar. We could keep the best and then showcase for example a Streaming or Tabulator app.

image

File: doc/index.md [hunk 1]
```diff
@@ -1,16 +1,74 @@
...
+
+In addition to these core features, Panel provides a number of **advanced features** that make it a unique and powerful tool for building data apps and dashboards. For example, **you can develop server-, pyodide-, and pyscript-backed applications, wrap your applications in polished templates for final deployment, and add authentication to your applications using the built-in OAuth providers**.
+
+Whether you're a domain expert, data scientist, software developer, or anyone in between, Panel provides a flexible and powerful platform for building interactive data apps and dashboards. With its broad range of features and support for multiple visualization libraries and development environments, **Panel is a unique and powerful tool that can help you bring your data to life**.
```
Comment (philippjfr) [inline doc/index.md:13]: Overall I like this description, one thing I don't think it conveys is that it allows building more complex applications than many of our competitors.

Comment (MarcSkovMadsen) [reply doc/index.md:13]: How would you formulate that?

**Comment by MarcSkovMadsen** [comment]
The Featured Videos are in.

To improve I would

- Rename it to Community Resources
- Make the images work
- Give the panels a title

image

**Comment by MarcSkovMadsen** [comment]
Maybe there should be a Featured tweets section too? And Featured Blog Posts too?

File: doc/index.md (hunk 1 above)
Comment (MarcSkovMadsen) [reply doc/index.md]: I've updated the home page. Need a new review on this section.

**Comment by MarcSkovMadsen** [comment]
I've updated the home page and just deployed it. I would need some review and feedback from you  @philippjfr and @droumis to get the energy and ideas for further improvements.

You can check out the improvements at https://pyviz-dev.github.io/panel/ once the action has finished. You might have to hard refresh your browser (I do) to see the changes.

You can check out what more could/ should be done in the top post. Thanks in advance for your feedback.

**Comment by MarcSkovMadsen** [comment]
I tried making the Demo App width responsive. But I could not. The tricks for solving layout issues pre Panel 1.0 seems not to work with Panel 1.0.

image

**Comment by MarcSkovMadsen** [comment]
The demo app looks so much better when it can stretch_width on the plot. But I simply could not get it working satisfactorily. The plot would not stretch when embedded.

The below is an example of a "wider" fixed size demo app.

image

File: doc/index.md [hunk 2]
```diff
@@ -1,25 +1,213 @@
+# Panel: The powerful data app solution for Python
```
Comment (MarcSkovMadsen) [inline doc/index.md:1]: Thinking about it. We could also use the header "More than a powerful data app solution for Python". Because Panel is not only for data apps. But its also a collection of carefully selected components and carefully crafted apis for working with data exploration and data visualization in notebooks and IDEs.

**Comment by philippjfr** [comment]
Some comments and quick impressions at a first glance from top to bottom.

1. I find the selection here a little odd, mixing the logo with a video and two app examples seems odd to me:

Screen Shot 2023-04-23 at 11 32 56

2. The descriptions seems like too much text to me, I don't think anyone is going to read that much.

3. The example looks good, and I like the re-org into APP, CODE, and DESCRIPTION tabs. I'm not fully convinced the undersaturated colors are an improvement.

The remaining sections all look great to me. Overall really great work but I do think the initial impression can be improved:

Screen Shot 2023-04-23 at 11 38 00

**Comment by MarcSkovMadsen** [comment]
Regarding

1. I sort of also think the description is too much, but we have to keep in mind it will be read by many types of readers. Some that will copy and distribute  the text. For example architects doing due diligence or AIs training to understand our solution.

I can try to minimize of @droumis thinks so too.

**Comment by MarcSkovMadsen** [comment]
Regarding

1. I wont object to removing the logo. It was just there already, so I tried to fit it in.

**Comment by MarcSkovMadsen** [comment]
Regarding 3. I'm just not so fond of the default HoloViews/ Bokeh Colorado. Together with the square and triangle you seldom see in business this makes it look a bit outdated to me. I tried to experiment with color maps and settled on the colorblind color map. To me its more aestetically pleasing, and for me its easier to focus on a single cluster. But i guess it individual.

**Comment by MarcSkovMadsen** [comment]
I've addressed comments 1. and 2. The logo is gone and the description is shorter.

image

You are very welcome to make a better image for the VideoStream app. But the one you currently have of your self is not very good because 1) It shows the Fast templates css issues 2) The ratio between height and width is not good 3) You don't have the  app header as the rest of the apps.

**Comment by philippjfr** [comment]
>I'm just not so fond of the default HoloViews/ Bokeh color map

The hvPlot and HoloViews color cycles are different and we have for a long time agreed that HoloViews has to change. The hvPlot one (which was used originally) was chosen deliberately because it mirrors the pandas default color cycle. The problem with the one you chose is that the colors are too similar:

Screen Shot 2023-04-23 at 21 33 17

Having yellow and orange be adjacent provides far too little contrast and having two blues in the first 5 colors is also less than ideal. I'll try to pick an alternative that's a little less saturated but maintains sufficient contrast.

**Comment by Hoxbro** [comment]
I think the home page is still very busy to look at. I think it should have almost no blocks of text, and I would put the video/showcase further down the page. They are good, but not the first thing a new user should see. 

The first thing should be an overview of what Panel is with bullet points that clearly convey the message at a quick glance. Maybe something like what Bokeh does:
image

Then the videos/showcase can be shown.

I think it would be great also to have more visuals when talking about the Holoviz ecosystem (and maybe also other things), something like what Dask does:
image

These are just my thoughts about this; I could have overlooked previous communication about this :upside_down_face:

**Comment by maximlt** [comment]
My 2cents on this. I feel that trying to design the home page so that it's both a landing page and the home documentation page makes it difficult to make the right decisions.

Presumably Panel (and the HoloViz sites) will get a proper/catchy landing page. I don't know if the efforts made to redesign in depth the current home page is worth it.

**Comment by MarcSkovMadsen** [comment]
I understand that you want to seperate the "marketing" and the "documentation". I wish I had known that before starting this PR. But at least it helped make this clear.

I just want the situation where I get the feedback from people I talk to that they don't "get" Panel. And they prefer to try using the alternatives first because they look "easier", "better" or more "fit for purpose". That is the experience I have big time right now. You can also see in the download numbers that Data App frameworks are growing massively. But we are doing the opposite.

**Comment by droumis** [comment]
I think there are more important things to focus on right now, but here are my two cents...

There's a lot to discuss and I think we should focus on what we could reasonably achieve and collectively live with prior to 1.0. We can always return to some ideas in the future. 

1. **Featured Dashboards**: If we have a row of featured dashboard cards at the top, there shouldn't be anything in those cards except inspiring, flawless, performant, and reliable dashboards. Later, when we have time to make a short intro video, it could go somewhere on this page, but not be grouped with the dashboards. Let's reallocate homepage-design-time to debugging and improving the featured dashboards. Also, integrate any descriptive text into a single section.

2. **Description**: This should be cut down a lot; probably just 3-4 short bullet points. When we have more time, we can try to use visuals that convey the information instead of words. In general, we should reconsider the use of cliches like 'you know and love'.

3. **Inline Example App**: The penguin app widgets and description tab are an improvement, but the color scheme still feels old. Let's just build from the colors that are already on the page to make it look consistent. Right now that's Panel-green, blue in the headings, and grey in the nav bar.

4. **Navigation Cards**: I don't know what the 'Learning Resources' cards achieve that the sidenav does not. There should be a single funnel for new users - going to the getting started page.

5. **HoloViz**: Let's mention HoloViz in about 5 words or less. It doesn't need a whole section or to mention other libraries by name - that's what HoloViz.org is for.

6. **Featured Resources**: So many videos, blog posts, and tweets make the page very busy. I do think we could have lists of community resources, but not on the homepage. Maybe we can create a new high-level page for them.

7. **Useful Links**: There's a lot here that is redundant (links to twitter, discord, github, etc), and in general this is too wordy. I don't think we need this section on the homepage.

8. **Contributing**: The contributing guide should be on a separate page under 'For Developers' (maybe rename 'For Developers' to 'For Contributors'). Remove from homepage.

9.  **Sponsors**: I think this is up to @philippjfr to determine what to include here, if anything.

___

Grabbing some points from #2171:
- Horizontal Logo: Sure, maybe we don't need both the horizontal logo in the topnav as well as a square logo at the top of the homepage.

- **Move first text lines up**: I disagree, I think the descriptive text should be condensed further and stay together, as mentioned above.

- **Make images bigger**: Sure, use less than 5 apps that make the cards bigger.

Everything else from #2171 is either not about the homepage or has already been handled

**Comment by maximlt** [comment]
> I understand that you want to separate the "marketing" and the "documentation". I wish I had known that before starting this PR. But at least it helped make this clear.

There's a recent initiative to produce a landing page for Panel, I though you were aware of it, if not there was a communication glitch somewhere.

@droumis I agree with each and everyone of your comments and suggestions above 👍

**Comment by MarcSkovMadsen** [comment]
Feedback for @droumis  comments

1.-3. Fine
4. I actually like the cards. They are also there in Pandas and Matplotlib so they are recognizable.
5. Somewhere you have to explain HoloViz. Being a well integrated with rest of HoloViz is a selling point. Without having an understanding of HoloViz its very hard to understand Panel. What are those reactive parameters? Why the big focus on using hvPlot or HoloViews in the examples?
6. Fine
7. The feedback I have from users is that they overlook the small icons in the top, right corner. And when they see them they don't understand what they symbolize or where they should go. I strongly believe users need to be introduced to where to go. Could alternatively be part of the getting started guide. Or an alternative page. You also need Google and ChatGPT to be able to find and understand this section.
8. Panel has a very low number of Github stars. And it matters for the acceptance of Panel. You need to do something actively to get them. What I am told is that you get them when you ask your users to give them. Users don't just jump somewhere and give them to you. I would strongly recommend the ask being there, somewhere very visible.
9. Sponsors information should be somewhere. That is what you look at when you evaluate the framework. Is it supported? Its a quality stamp.

**Comment by MarcSkovMadsen** [comment]
Closing as it did not make it.
